src/provenaclient/modules/submodules/datastore_io_submodule.py

```diff
@@ -82,6 +92,53 @@ def __init__(self, auth: AuthManager, config: Config, datastore_client: Datastor
         # Clients related to the datastore scoped as private.
         self._datastore_client = datastore_client
 
+    async def _create_s3_path(self, dataset_id: str, access_type: AccessEnum) -> S3Path:
+        """This helper function creates an S3 URI in PATH format by ingesting 
+        the dataset id and access type (read, write). 
+
+        Parameters
+        ----------
+        dataset_id : str
+            The ID of the dataset to download files for - ensure you have the right access.
+        access_type : AccessEnum
+            The access type required (Read or Write)
+
+        Returns
+        -------
+        S3Path
+            S3Path instance that represent a path in S3 with filesystem path semantics.
+        """
+
+        dataset_information = await self._datastore_client.fetch_dataset(id=dataset_id)
+
+        # Fetch the dataset information
+        dataset_information = await self._datastore_client.fetch_dataset(
+            id=dataset_id
+        )
+        # Get the S3 location
+        assert dataset_information.item is not None, f"Expected non None item from dataset fetch, details: {dataset_information.status.details}."
+        s3_location = dataset_information.item.s3
+
+        if access_type == AccessEnum.READ:
+            creds = await self._datastore_client.generate_read_access_credentials(
+                read_access_credentials=CredentialsRequest(
+                    dataset_id=dataset_id,
+                    console_session_required=False
+                )
+            )
+
+        elif access_type == AccessEnum.WRITE:
```

Comment:
I have added an else condition as a safeguard. 

The mapping approach is interesting, but wouldn't it be too over-engineered for a small set of access cases (read and write)? However, I'm happy to implement it if you think otherwise.

Reply:
Yeah I think it probably can be over doing it for this example.